docs/src/account.md

```diff
@@ -83,6 +83,19 @@ The [storage](https://docs.rs/miden-objects/latest/miden_objects/account/struct.
 - **`StorageSlot::Value`:** Contains 32 bytes of arbitrary data.
 - **`StorageSlot::Map`:** Contains a [StorageMap](https://docs.rs/miden-objects/latest/miden_objects/account/struct.StorageMap.html), a key-value store where both keys and values are 32 bytes. The slot's value is a commitment to the entire map.
 
+#### StorageMap
+
+A `StorageMap` is a key-value store implemented as a sparse Merkle tree (SMT) of depth 64. This allows an account to store a much larger amount of data than would be possible using only the account's storage slots. The root of the underlying SMT is stored in a single account storage slot, and each map entry is a leaf in the tree. When retrieving an entry (e.g., via `account::get_map_item`), its inclusion is proven using a Merkle proof.
+
+Key properties of `StorageMap`:
+
+- **Efficient, scalable storage:** The SMT structure enables efficient storage and proof of inclusion for a large number of entries, while only storing the root in the account's storage slot.
+- **Partial presence:** Not all entries of the map need to be present at transaction execution time to access or modify the map. It is sufficient if only the accessed or modified items are present in the advice provider.
+- **Key hashing:** Since map keys are user-chosen and may not be uniformly distributed, keys are hashed before being inserted into the SMT. This ensures a more balanced tree and mitigates efficiency issues due to key clustering. The original keys are retained in a separate map, allowing for introspection (e.g., querying the set of stored original keys for debugging or explorer scenarios).
+- **Redundancy for introspection:** Retaining original keys in a separate map introduces some redundancy, but enables useful features such as listing all stored keys.
```

Comment:
I'm not sure if I would list this as a "key property" of the storage map. Instead I would just add this information under the previous "key hashing" bullet point.